src/test/java/jenkins/branch/CustomOrganizationFolderDescriptorTest.java

```diff
@@ -76,35 +76,41 @@ public void removeIntegrationTestMocks() throws Exception {
     
     @Test
     public void noNavigatorNoFactoryInstalled() throws Exception {
-        assertEquals(1, ExtensionList.lookup(SCMNavigatorDescriptor.class).size());
-        assertEquals(SingleSCMNavigator.DescriptorImpl.class, ExtensionList.lookup(SCMNavigatorDescriptor.class).get(0).getClass());
-        assertEquals(0, ExtensionList.lookup(MultiBranchProjectFactoryDescriptor.class).size());
+        assertThat(ExtensionList.lookup(SCMNavigatorDescriptor.class),
+                hasItem(ExtensionList.lookupSingleton(SingleSCMNavigator.DescriptorImpl.class)));
+        // Ensure none of our mock factories sneaked into this test (though other MultiBranchProjectFactory impls may exist on the classpath)
+        assertThat(ExtensionList.lookup(OrganizationFolderTest.MockFactoryDescriptor.class), is(empty()));
         assertThat(newItemTypes(), containsInAnyOrder());
     }
 
     @Test
     public void someNavigatorNoFactoryInstalled() throws Exception {
-        assertEquals(2, ExtensionList.lookup(SCMNavigatorDescriptor.class).size());
-        assertEquals(0, ExtensionList.lookup(MultiBranchProjectFactoryDescriptor.class).size());
+        assertThat(ExtensionList.lookup(SCMNavigatorDescriptor.class),
+                hasItems(ExtensionList.lookupSingleton(SingleSCMNavigator.DescriptorImpl.class), ExtensionList.lookupSingleton(SomeNavigatorNoFactoryInstalledDescriptor.class)));
+        // Ensure none of our mock factories sneaked into this test (though other MultiBranchProjectFactory impls may exist on the classpath)
+        assertThat(ExtensionList.lookup(OrganizationFolderTest.MockFactoryDescriptor.class), is(empty()));
         assertThat(newItemTypes(), containsInAnyOrder());
     }
     @TestExtension("someNavigatorNoFactoryInstalled")
     public static class SomeNavigatorNoFactoryInstalledDescriptor extends MockNavigatorDescriptor {}
 
     @Test
     public void noNavigatorSomeFactoryInstalled() throws Exception {
-        assertEquals(1, ExtensionList.lookup(SCMNavigatorDescriptor.class).size());
-        assertEquals(SingleSCMNavigator.DescriptorImpl.class, ExtensionList.lookup(SCMNavigatorDescriptor.class).get(0).getClass());
-        assertEquals(1, ExtensionList.lookup(MultiBranchProjectFactoryDescriptor.class).size());
+        assertThat(ExtensionList.lookup(SCMNavigatorDescriptor.class),
+                hasItem(ExtensionList.lookupSingleton(SingleSCMNavigator.DescriptorImpl.class)));
```

Comment:
Again I think you are missing the check that there is no `MockNavigatorDescriptor`.